astropy/modeling/parameters.py

```diff
@@ -345,26 +283,22 @@ def __setitem__(self, key, value):
                 oldvalue[key] = value
             except IndexError:
                 raise InputParameterError(
-                    "Input dimension {} invalid for {!r} parameter with "
-                    "dimension {}".format(key, self.name, n_models))
+                    "Input dimension {0} invalid for {1!r} parameter with "
+                    "dimension {2}".format(key, self.name, value.shape[0]))  # likely wrong
 
     def __repr__(self):
         args = f"'{self._name}'"
-        if self._model is None:
-            if self._default is not None:
-                args += f', default={self._default}'
-        else:
-            args += f', value={self.value}'
+        args += ', value={0}'.format(self.value)
 
         if self.unit is not None:
-            args += f', unit={self.unit}'
```

Comment:
a rebase went wrong here and a few other places, I think (around those merge commits). this is not at all blocker for merging, but I'm happy to fix it before it gets merged.